sh_scrapy/settings.py

```diff
@@ -95,12 +101,21 @@ def _load_addons(addons, settings, merged_settings, priority=0):
             raise
         skey = _get_component_base(settings, addon['type'])
         components = settings[skey]
-        path = update_classpath(addon_path)
-        components[path] = addon['order']
+        _update_component_order(components, addon_path, addon['order'])
         merged_settings.set(skey, components)
         merged_settings.setdict(addon['default_settings'], priority)
 
 
+def _merge_with_keeping_order(settings, updates):
+    for setting_key, value in updates.items():
+        if not isinstance(value, dict):
+            settings.set(setting_key, value, priority='cmdline')
+            continue
+        components = settings[setting_key]
+        for path, order in value.items():
+            _update_component_order(components, path, order)
```

Comment:
Just drop a comment - if something will change in future (e.g. if someone will modify copy of the dict settings instead of mutating them directly) it'll be easier to debug this.